packages/consumption/src/modules/attributes/AttributesController.ts

```diff
@@ -1049,6 +1081,62 @@ export class AttributesController extends ConsumptionBaseController {
         }
     }
 
+    private async transmitDefault(attribute: LocalAttribute): Promise<void> {
+        if (!attribute.default) return;
+
+        const valueType = attribute.content.value.constructor.name;
+        const query = {
+            $and: [
+                {
+                    [`${nameof<LocalAttribute>((c) => c.content)}.value.@type`]: valueType
+                },
+                {
+                    [nameof<LocalAttribute>((c) => c.succeededBy)]: undefined
+                },
+                {
+                    [nameof<LocalAttribute>((c) => c.id)]: { $ne: attribute.id.toString() }
+                }
+            ]
+        };
+
+        const defaultCandidates = await this.getLocalAttributes(query);
+        if (defaultCandidates.length === 0) return;
+
+        defaultCandidates[0].default = true;
+        await this.updateAttributeUnsafe(defaultCandidates[0]);
+    }
+
+    public async changeDefaultRepositoryAttribute(newDefaultAttribute: LocalAttribute): Promise<LocalAttribute> {
```

Comment:
The `ChangeDefaultRepositoryAttributeUseCase` which calls this `changeDefaultRepositoryAttribute` function checks if `newDefaultAttribute` is a RepositoryAttribute. Do we want to replicate this check or do we want to transfer it from the use case to the function within the `AttributesController`?

And just out of interest: Public methods of the Controllers (within consumption, for example) can't be used by integrators but can be used by us? So your motivation of adding a check for the `changeDefaultRepositoryAttribute` function is, for example, that it causes _our_ coding to be less error-prone?

Reply:
Yes, integrators can only use use cases. But we can internally use public controller functions, e.g. in RequestItemProcessors.

Reply:
I now duplicated the check, such that if the function is called from the runtime use case, a runtime error will be thrown instead of the consumption error.